datamol/mol.py

```diff
@@ -154,38 +156,64 @@ def to_neutral(mol: Chem.rdchem.Mol) -> Optional[Chem.rdchem.Mol]:
 
 
 def sanitize_mol(
-    mol: Chem.rdchem.Mol, charge_neutral: bool = False, sanifix: bool = True
+    mol: Chem.rdchem.Mol,
+    charge_neutral: bool = False,
+    sanifix: bool = True,
+    verbose: bool = True,
 ) -> Optional[Chem.rdchem.Mol]:
-    """Sanitize molecule and fix common errors.
+    """An augmented version of RDKit `sanitize=True`. It uses a
+    mol-SMILES-mol conversion to catch potential aromaticity errors
+    and try to fix aromatic nitrogen (using the popular sanifix4 script).
+    Optionally, it can neutralize the charge of the molecule.
+
+    Note #1: Only the first conformer (if present) will be preserved and
+    a warning will be displayed if more than one conformer is detected.
 
-    Warning:
-        The procedure includes a SMILES conversion to avoid accasional aromaticity
-        errors. In consequence, all the properties and the conformers will be lost.
+    Note #2: The molecule's properties will be preserved but the atom's
+    properties will be lost.
 
     Args:
         mol: a molecule.
         charge_neutral: whether charge neutralization should be applied.
         sanifix: whether to run the sanifix from James Davidson
             (sanifix4.py) that try to adjust aromatic nitrogens.
+        verbose: Whether displaying a warning about multiple conformers.
 
     Returns:
         mol: a molecule.
     """
     if mol is None:
         return mol
 
+    # Extract properties.
+    original_mol = copy_mol(mol)
+    properties = original_mol.GetPropsAsDict()
+
     if charge_neutral:
         mol = to_neutral(mol)
 
     if sanifix:
         mol = _sanifix4.sanifix(mol)
 
-    if mol:
+    if mol is not None:
+
+        # Detect multiple conformers
+        if verbose and mol.GetNumConformers() > 1:
+            logger.warning(
+                f"The molecule contains multiple conformers. Only the first one will be preserved."
+            )
+
+        # Try catch to avoid occasional aromaticity errors
         try:
-            # Try catch to avoid occasional aromaticity errors
-            return to_mol(dm.to_smiles(mol), sanitize=True)  # type: ignore
+            # `cxsmiles` is used here to preserve the first conformer.
+            mol = to_mol(dm.to_smiles(mol, cxsmiles=True), sanitize=True)  # type: ignore
```

Comment:
On the above, `add_hs=True` for `to_mol` does not work as expected.


![image](https://user-images.githubusercontent.com/5290110/121571522-229dcb80-c9f1-11eb-8b1a-36affc806135.png)

Reply:
Yup good catch here. I have added `add_hs` to `sanitize_mol`. To make it work I had to add `Chem.AddHs(_mol, explicitOnly=explicit_only, addCoords=True)` to `dm.to_mol`.

Can you think of any issue having `addCoords` always `True`?

Reply:
```python
mol = dm.to_mol("CC(=[O+])OC1=NC=C2C=NNC2=C1C([O-])=O", add_hs=True)  
mol = dm.conformers.generate(mol, n_confs=1, add_hs=False)
new_mol = dm.sanitize_mol(mol, charge_neutral=False, add_hs=True)
```

![image](https://user-images.githubusercontent.com/528003/121604042-d6b24d00-ca17-11eb-959e-fcb59aaf5b94.png)

Reply:
yes, 
```Chem.AddHs(_mol, explicitOnly=explicit_only, addCoords=True)``` works (see https://github.com/datamol-org/datamol/pull/41#discussion_r649393291 ). However, the hydrogen position are not nearly the same (see original conformer or second figure of the comment above).  I don't think that's a big issue.

I tried `allBondsExplicit=True, allHsExplicit=False` in `MolToCXSmiles` before and it did not change anything. So that's probably the best we could do, without using atom mapping and trying to remap hydrogen positions (needless complexity imo).

Reply:
> However, the hydrogen position are not nearly the same

Yup I noticed that but I think it's fair and fine. `addCoords` is probably using some kinds of deterministic positioning system that will be likely different than a random embedding without energy minimization.

Yup let's avoid atom mapping if we can here as those functions can potentially be called during a large scale screening where you want to go fast!